indico/web/client/js/react/components/principals/Search.jsx

```diff
@@ -221,30 +221,86 @@ const searchFactory = config => {
     withEventPersons,
     eventId,
   }) => {
-    const [result, setResult] = useState(null);
+    const [loading, setLoading] = useState(false);
+    const [result, _setResult] = useState(null);
+    const lastResult = useRef(null);
+    const setResult = value => {
+      lastResult.current = result;
+      _setResult(value);
+    };
+    const handleSearch = async data => {
+      setLoading(true);
+      try {
+        await runSearch(data, setResult, withEventPersons, eventId);
+      } finally {
+        setLoading(false);
+      }
+    };
+    const favoriteResults = Object.values(favorites);
```

Comment:
this breaks on pages that do not use favorites (e.g. the favorite user management page).

This avoids it:

```suggestion
    const favoriteResults = favorites ? Object.values(favorites) : [];
```

It would be nice to show something like "n favorite users" instead of "n users found" when we show the favorites initially... (and if there are no favorites, something else)

Reply:
Thanks, hadn't realized this could be null.
I thought about something like "Suggested users" but not sure it's worth it since `noResultsText` is actually a prop.

Reply:
You could add `initialResultsText` (for the favorites - in case of groups we don't have anything there so we'd just leave it empty) and `notSearchedText` that's shown when there are no initial results nor a search.

Reply:
I tried using a different text for the initial results, but having a different string doesn't look great as there's still a lot of white space. 
As an alternative, I tried to initially hide the results column if there are neither results nor favourites - but only if the search is "pristine". How does it look to you?

The only part left to address is `initialResultsText` **when the favourites are being shown**, where I can have something as "n favorite users".

Reply:
user search is currently broken (undefined results in SearchResults), but besides the fact that it currently shows "no results found" while searching (ldap group search without "exact match" takes a while) I kind of like it..

Reply:
works pretty nicely now, just one small thing: while the right side is disabled because of a running search, the widgets there are still active - ie i can select users and get tooltips when hovering the "user exists in event" icon

Reply:
I disabled the pointer-events. Not exactly sure if it's worth changing the faded "no results found" while the next search state is still loading since I'm not sure what to set there instead, unless I got rid of it.

Reply:
I like it the way it is now :)

You could add `user-select: none;` though; it doesn't really matter but being able to select content by clicking and dragging the cursor on a selected area feels a bit strange.